modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainerCluster.java

```diff
@@ -0,0 +1,122 @@
+
+    public KafkaContainerCluster(int brokersNum, int internalTopicsRf) {
+        this(CONFLUENT_PLATFORM_VERSION, brokersNum, internalTopicsRf);
+    }
+
+    public KafkaContainerCluster(String confluentPlatformVersion, int brokersNum, int internalTopicsRf) {
+        if (brokersNum < 0) {
+            throw new IllegalArgumentException("brokersNum '" + brokersNum + "' must be greater than 0");
+        }
+        if (internalTopicsRf < 0 || internalTopicsRf > brokersNum) {
+            throw new IllegalArgumentException("internalTopicsRf '" + internalTopicsRf + "' must be less than brokersNum and greater than 0");
+        }
+
+        this.brokersNum = brokersNum;
+        this.network = Network.newNetwork();
+
+        this.zookeeper = new GenericContainer("confluentinc/cp-zookeeper:" + confluentPlatformVersion)
+            .withNetwork(network)
+            .withNetworkAliases("zookeeper")
+            .withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(KafkaContainer.ZOOKEEPER_PORT));
+
+        this.brokers = IntStream
+            .range(0, this.brokersNum)
+            .mapToObj(brokerNum ->
+                new KafkaContainer(confluentPlatformVersion)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerNum)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + KafkaContainer.ZOOKEEPER_PORT)
+                    .withEnv("KAFKA_BROKER_ID", brokerNum + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_MIN_ISR", internalTopicsRf + "")
+            )
+            .collect(Collectors.toList());
+    }
+
+    public Network getNetwork() {
+        return this.network;
+    }
+
+    public GenericContainer getZooKeeper() {
+        return this.zookeeper;
+    }
+
+    public Collection<KafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(KafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    private Stream<GenericContainer> allContainers() {
+        Stream<GenericContainer> genericBrokers = this.brokers.stream().map(b -> (GenericContainer) b);
+        Stream<GenericContainer> zookeeper = Stream.of(this.zookeeper);
+        return Stream.concat(genericBrokers, zookeeper);
+    }
+
+    @Override
+    @SneakyThrows
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(b -> (Startable) b);
+        Startables.deepStart(startables).get(60, SECONDS);
+
+        Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> Stream.of(this.zookeeper)
```

Comment:
I believe it can be simplified to `() -> clusterBrokersIDs().length == brokersNum` (where `clusterBrokersIDs` returns an array (or list) of broker IDs (or, if `execInContainer` is possible, can be inlined since it becomes a one liner)

Reply:
The container command returns a string (internally it's stored as a string, I guess, i.e. `1,2,3`).  I could inline a shell command to split or count the string instead if that's what you prefer.